src/components/settings-modal/settings-modal.test.js

```diff
@@ -73,5 +74,11 @@ describe('SettingsModal', () => {
       type: 'TOGGLE_IS_PRETTY_NAME',
       isPrettyName: false,
     });
+
+    mapDispatchToProps(dispatch).onToggleShowDatasetPreviews(false);
+    expect(dispatch.mock.calls[3][0]).toEqual({
+      type: 'UPDATE_USER_PREFERENCES',
+      payload: { preferences: { showDatasetPreviews: false } },
```

Comment:
UPDATE_USER_PREFERENCES should take an object preferences. The key preferences is not needed. The object is a preferences object as a whole

```
{ 
  showDatasetPreviews: false 
}
```